compaheuiler/src/c_bigint_runtime.c

```
@@ -0,0 +1,125 @@
+static int64_t wrapping_rem(int64_t a,int64_t b){if(a==INT64_MIN&&b==-1)return 0;int64_t r=a%b;if(r!=0&&(r<0)!=(b<0))r+=b;return r;}
+static int64_t int_div(int64_t a,int64_t b,int bm){if(!bm)return b?wrapping_div(a,b):0;if(b==1)return 1;if((a&b&1)!=0)return tag_small(wrapping_div(a>>1,b>>1));return cbig_div(a,b);}
+static int64_t int_rem(int64_t a,int64_t b,int bm){if(!bm)return b?wrapping_rem(a,b):0;if(b==1)return 1;if((a&b&1)!=0)return tag_small(wrapping_rem(a>>1,b>>1));return cbig_rem(a,b);}
+
+static char _outbuf[16384]; static size_t _outpos=0;
+static void _flush(void){if(_outpos){fwrite(_outbuf,1,_outpos,stdout);_outpos=0;}}
+static void _emit(char c){_outbuf[_outpos++]=c;if(_outpos>=sizeof(_outbuf))_flush();}
+static void _emit_u8(uint8_t c){_emit((char)c);}
+static void write_i64(int64_t n){uint64_t u;if(n<0){_emit('-');u=(uint64_t)(-(n+1))+1;}else u=n;char b[20];int i=20;do{b[--i]='0'+u%10;u/=10;}while(u);while(i<20)_emit(b[i++]);}
+static void write_num(int64_t v,int bm){if(!bm){write_i64(v);return;}if(v&1){write_i64(v>>1);return;}cbig_write_num(v,_emit_u8);}
+static void write_char(int64_t v,int bm){uint32_t c=(uint32_t)int_to_i64(v,bm);if(c<=0x7f)_emit((char)c);else if(c<=0x7ff){_emit(0xc0|(c>>6));_emit(0x80|(c&63));}else if(c<=0xffff){_emit(0xe0|(c>>12));_emit(0x80|((c>>6)&63));_emit(0x80|(c&63));}else if(c<=0x10ffff){_emit(0xf0|(c>>18));_emit(0x80|((c>>12)&63));_emit(0x80|((c>>6)&63));_emit(0x80|(c&63));}}
+static int64_t read_num(int bm){char b[64];if(!fgets(b,sizeof(b),stdin))return int_lit(0,bm);return int_lit((int64_t)atoll(b),bm);}
```

Cppcheck (2.21.0)

[style] The function 'read_num' is never used.

(unusedFunction)